integrations/uploader_downloader_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integrations
+
+import (
+	"context"
+	"net/url"
+	"os"
+	"path/filepath"
+	"testing"
+
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/models/unittest"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/modules/util"
+	"code.gitea.io/gitea/services/migrations"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUploaderDownloader(t *testing.T) {
+	onGiteaRun(t, func(t *testing.T, u *url.URL) {
+		AllowedDomains := setting.Migrations.AllowedDomains
+		setting.Migrations.AllowedDomains = "loopback"
+		AllowLocalNetworks := setting.Migrations.AllowLocalNetworks
+		setting.Migrations.AllowLocalNetworks = true
+		setting.Database.LogSQL = true
+		AppVer := setting.AppVer
+		setting.AppVer = "1.15"
```

Comment:
Do we really need to change these settings?
```suggestion
```

Reply:
> Relaying from Loïc Dachary :

@wxiaoguang thanks for the quick review :-)

>  setting.Database.LogSQL

It is not necessary indeed and will be removed, good catch.
> setting.AppVer = "1.15"

There is [no default value for AppVer](https://github.com/go-gitea/gitea/blob/main/modules/setting/setting.go#L70) and the test will fail with the following error if it is not set.

```bash
   testlogger.go:78: 2021/12/30 10:11:27 ...ers/common/logger.go:30:1() [I] Completed GET /api/v1/version 200 OK in 17.200939ms
    dump_restore_test.go:68: 
        	Error Trace:	dump_restore_test.go:68
        	            				git_helper_for_declarative_test.go:113
        	            				git_helper_for_declarative_test.go:108
        	            				git_helper_for_declarative_test.go:112
        	            				dump_restore_test.go:28
        	Error:      	Received unexpected error:
        	            	Malformed version: 
        	Test:       	TestDumpRestore
```